### PR TITLE
Update top nav input box action button icon depends on input

### PR DESCRIPTION
**Pull request by PikachuEXE**
**Pull Request Type**
Please select what type of pull request this is:
- [x] Feature Implementation

**Related issue**
It's [a comment/feedback in another PR](https://github.com/FreeTubeApp/FreeTube/pull/1669#issuecomment-918015016)

**Description**
Before: Top nav input box action button icon is always "arrow right"
After: Action button icon changes depend on input text "looks like a youtube URL" or not
- Initial: icon = "search"
- Looks like YT URL: icon = "arrow right"
- Otherwise: icon = "search"

Also implements
["Having the disappear animation of the Clear Text button also occur on appear/re-appear for visual consistency."](https://github.com/FreeTubeApp/FreeTube/pull/1669#issuecomment-924003650)

**Screenshots (if appropriate)**
Icon change:
Screenshot 2021-09-23 at 3 04 46 PM
Screenshot 2021-09-23 at 3 04 30 PM

Button animation change:

https://user-images.githubusercontent.com/1018543/134470580-c3f004b7-4b2a-4dff-8352-fa35f41797b5.mov

"Update action button to look & act as disabled when input text is empty"

https://user-images.githubusercontent.com/1018543/134834657-3931df76-608a-4af7-91d6-7cccc3561600.mov

**Testing (for code that is not small enough to be easily understandable)**
- Enter Youtube URLs (e.g. Video, Playlist, Channel), look at icon changed
- Clear text via different methods after Youtube URLs entered
- Enter non Youtube URLs (any text)
- Clear text via different methods after text entered

**Desktop (please complete the following information):**
- OS: MacOS
- OS Version: 11.6
- FreeTube version: based on b1a310ef09cdad6f2eda5b65a544e5b2e0880648

**Additional context**

### CODE REVIEW

**Review by peepopoggers — CHANGES_REQUESTED** [review]
Tested this locally and it works great.
In my opinion it should accept all URL's for it to display arrow as this does not account for invidious links / future additions.

**Comment by PikachuEXE** [comment]
I now instead updated it to use `getYoutubeUrlInfo`
And update the icon according to the `urlType`

### Example URLs 

#### Video with Playlist
- https://www.youtube.com/watch?v=_qmCjsIU2fc&list=PLzFTGYa_evXjIHj3YW0Xp5PiVzjpd-IBG
- https://yewtu.be/watch?v=_qmCjsIU2fc&list=PLzFTGYa_evXjIHj3YW0Xp5PiVzjpd-IBG

#### Playlist
- https://youtube.com/playlist?list=PLzFTGYa_evXjrYYxeY6HAQyqnkJju2t84
- https://yewtu.be/playlist?list=PLzFTGYa_evXjrYYxeY6HAQyqnkJju2t84

#### Channel
- https://www.youtube.com/channel/UCz7eke4JGlbtc11_6mmA7ew
- https://yewtu.be/channel/UCz7eke4JGlbtc11_6mmA7ew

#### Search (Only Youtube URL supported by FT now)
- https://www.youtube.com/results?search_query=pikachu

File: src/renderer/components/ft-input/ft-input.js [hunk 1]
```diff
@@ -118,18 +128,84 @@ export default Vue.extend({
...
+            case 'hashtag': {
+              // TODO: Implement a hashtag related view
+              // isYoutubeLink is already `false`
+              break
+            }
+
+            case 'channel': {
+              isYoutubeLink = true
+              break
+            }
```
Comment (jasonhenriquez) [inline src/renderer/components/ft-input/ft-input.js:185]:
```suggestion
            case 'video':
            case 'playlist':
            case 'search':
            case 'channel':
              isYoutubeLink = true
              break
            case 'hashtag':
              // TODO: Implement a hashtag related view
              // isYoutubeLink is already `false`
              break
```

**Comment by PikachuEXE** [comment]
@jasonhenriquez 
Updated

**Comment by peepopoggers** [comment]
What about youtube shorts?

**Comment by PikachuEXE** [comment]
@peepopoggers 
Is it supported by YT now?
I am only following `getYoutubeUrlInfo` here

**Comment by peepopoggers** [comment]
Test this:
https://www.youtube.com/shorts/MS82yTaKRjQ?feature=share

**Comment by peepopoggers** [comment]
#1605

**Comment by PikachuEXE** [comment]
https://www.youtube.com/shorts/MS82yTaKRjQ?feature=share
This PR seems working
image

**Review by peepopoggers — DISMISSED** [review]
:+1:

**Comment by PikachuEXE** [comment]
"Update action button to look & act as disabled when input text is empty"

https://user-images.githubusercontent.com/1018543/134834657-3931df76-608a-4af7-91d6-7cccc3561600.mov

@jasonhenriquez

File: src/renderer/components/ft-input/ft-input.css [hunk 2]
```diff
@@ -90,12 +90,16 @@
...
-  right: 0px;
+  right: 0;
   cursor: pointer;
   border-radius: 200px 200px 200px 200px;
   color: var(--primary-text-color);
 }
 
+.inputAction.disabled {
+  opacity: 50%;
+}
```
Comment (ChunkyProgrammer) [inline src/renderer/components/ft-input/ft-input.css:101]:
```suggestion
  cursor: unset;
}
```

**Comment by ChunkyProgrammer** [comment]
Maybe unset the background color on hover & active when disabled as well?

**Comment by PikachuEXE** [comment]
Get 1 more person to agree with you and it will be done
I am not sure if removing those effect would have negative effect on UX

**Comment by peepopoggers** [comment]
@ChunkyProgrammer can you explain what these changes you would propose would do in more detail?

**Comment by ChunkyProgrammer** [comment]
> @ChunkyProgrammer can you explain what these changes you would propose would do in more detail?

Sure,
If the search bar is empty then I think the search icon shouldn't react to css events like hovering and activated as the search functionality is disabled at that point. If you test the PR and click on the search icon, there's a visual indication that you clicked on (background-color changes) which gives the impression (to me at least) of the search being successful even if search is disabled.

**Comment by peepopoggers** [comment]
Okay, I agree with that functionality as searching for nothing does not make much sense.

**Comment by jasonhenriquez** [comment]
I second that.

**Comment by PikachuEXE** [comment]
Updated

https://user-images.githubusercontent.com/1018543/135020667-2bc8c733-bd8f-4dd3-9890-d785dbd23d8a.mov

File: src/renderer/components/ft-input/ft-input.css [hunk 3]
```diff
@@ -90,10 +90,15 @@
   position: absolute;
   padding: 10px 8px;
   top: 5px;
-  right: 0px;
+  right: 0;
   cursor: pointer;
```
Comment (ChunkyProgrammer) [inline src/renderer/components/ft-input/ft-input.css:94]: I'd move this line to .inputAction.enabled (line 100)

**Comment by PikachuEXE** [comment]
@ChunkyProgrammer 
Updated

https://user-images.githubusercontent.com/1018543/135192108-4b95c0e0-abd3-43f4-b50b-d44e6d0c3742.mov

**Review by peepopoggers — APPROVED** [review]
LGTM

**Review by ChunkyProgrammer — APPROVED** [review]

**Comment by PikachuEXE** [comment]
1 more to go!

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]
Press B, LGTM!

**Comment by PikachuEXE** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc 
You want to cancel the merge!?

**Pull request merged**